### PR TITLE
Get the correct result of a "simple" dialog in mahapps

### LINKED ISSUE #61: Showing a custom dialog does not return the correct result in Mahapps
### Steps to reproduce

1. call `var dialogResult = await IUIVisualizerService.ShowDialogAsync<ExampleDialogOkCancelViewModel>();`. Whereas the corresponding view of `ExampleDialogOkCancelViewModel` is
```c#
    public partial class ExampleDialogOkCancelView : SimpleDataWindow
    {
        public ExampleDialogOkCancelView()
            : base(DataWindowMode.OkCancel)
        {
            InitializeComponent();
        }
    }
```
2. In the dialog click the Cancel button

### Expected behaviour
The value of `dialogResult` is `false`

Action | Result
------------ | -------------
Cancel | false
Ok | true
Apply | none, the dialog does not close
Close | null (??)

### Actual behaviour
calling `IUIVisualizerService.ShowDialogAsync` in Mahapps (implementation `MahAppsUIVisualizerService.ShowWindowAsync(FrameworkElement window, bool showModal)`) always returns `true` for a view of the type `CustomDialog`

@GeertvanHorrik can you review the content of the table, especially the result value of the close action. I will implement it and make a PR

### PR DESCRIPTION
**Pull request by manne**
Fixes #61.

### Checklist

- [x] I have included examples or tests
- [ ] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file
- [x] I have cleaned up the commit history (use rebase and squash)

### Changes proposed in this pull request:

- Possibility to get the correct result of a "simple" dialog in mahapps

### CODE REVIEW

File: src/Examples/Orchestra.Examples.MahApps/Orchestra.Examples.MahApps.Shared/ViewModels/ControlsViewModel.cs [hunk 1]
```diff
@@ -19,14 +19,47 @@ public ControlsViewModel(IUIVisualizerService uiVisualizerService, IMessageServi
             _messageService = messageService;
 
             OnShowCustomDialogExecute = new TaskCommand(ShowCustomDialogExecuteAsync);
+            OnShowCloseDialogExecute = new TaskCommand(ShowCloseDialogExecuteAsync);
+            OnShowOkCancelDialogExecute = new TaskCommand(ShowOkCancelDialogExecuteAsync);
+            OnShowOkCancelApplyDialogExecute = new TaskCommand(ShowOkCancelApplyDialogExecuteAsync);
+        }
+
+        public TaskCommand OnShowCustomDialogExecute { get; private set; }
```
Comment (GeertvanHorrik) [inline src/Examples/Orchestra.Examples.MahApps/Orchestra.Examples.MahApps.Shared/ViewModels/ControlsViewModel.cs]: Commands should have a different name:

Command name: ShowCloseDialog
Method name: OnShowCloseDialogExecute

File: src/Orchestra.Shell/Orchestra.Shell.MahApps/Orchestra.Shell.MahApps.Shared/Windows/SimpleDataWindow.cs [hunk 2]
```diff
@@ -148,12 +149,12 @@ protected bool OnOkCanExecute()
         /// </summary>
         protected async Task OnOkExecuteAsync()
         {
-            if (!await ApplyChangesAsync())
+            if (!await ApplyChangesAsync().ConfigureAwait(false))
```
Comment (GeertvanHorrik) [inline src/Orchestra.Shell/Orchestra.Shell.MahApps/Orchestra.Shell.MahApps.Shared/Windows/SimpleDataWindow.cs]: Why ConfigureAwait(false)? We want it to return the call to the UI context?

File: src/Examples/Orchestra.Examples.MahApps/Orchestra.Examples.MahApps.Shared/ViewModels/ControlsViewModel.cs (hunk 1 above)
Comment (manne) [inline src/Examples/Orchestra.Examples.MahApps/Orchestra.Examples.MahApps.Shared/ViewModels/ControlsViewModel.cs]: i will rename the properties and methods

File: src/Orchestra.Shell/Orchestra.Shell.MahApps/Orchestra.Shell.MahApps.Shared/Windows/SimpleDataWindow.cs (hunk 2 above)
Comment (manne) [inline src/Orchestra.Shell/Orchestra.Shell.MahApps/Orchestra.Shell.MahApps.Shared/Windows/SimpleDataWindow.cs]: i will remove the `ConfigureAwait(false)` from the execute methods called by the commands

**Comment by GeertvanHorrik** [comment]
@manne were you able to make the changes?

**Comment by GeertvanHorrik** [comment]
@alexfdezsauco , please merge this PR + update the code according to my review.

**Comment by alexfdezsauco** [comment]
fixed

**Pull request merged**